### ISSUE TITLE
Support :has pseudo class 

### ISSUE THREAD

**Issue by cderv** [b-fuze/deno-dom#139]
This is about the pseudo class `:has` (https://developer.mozilla.org/en-US/docs/Web/CSS/:has#specifications) which currently return a DOMException

````typescript
import { DOMParser, Element } from "https://deno.land/x/deno_dom/deno-dom-wasm.ts";
const doc = new DOMParser().parseFromString(
  `
  <h1>Hello World!</h1>
  <p>Hello from <a href="https://deno.land/">Deno!</a></p>
`,
  "text/html",
);
const p = doc.querySelector("p");
const p2 = doc.querySelector("p:has(a)");
// returns Uncaught DOMException
````

The above works in JS in browser console.

**Comment by b-fuze** [comment]
I'll have to check if our upstream dependency NWSAPI has implemented it already; I think that they have and I need to vendor their newer release anyways since it fixes some long standing bugs (e.g. `:scope`)

**Comment by oles** [comment]
:has usage is becoming more and more common now - I'd love to see support for :has for deno-dom :heart:

**Comment by b-fuze** [comment]
Yeah, me too tbh. The problem currently is that nwsapi has bugs in its `:has` selector implementation... And while I'm interested in other engines like `css-select` from NPM I do not like the non trivial drop in performance that woupd result from that... So currently I'm thinking of forking nwsapi for deno-dom and fixing it myself and maybe the nwsapi authors can upstream my changes or fix it in the way they prefer when they get the time and then I can decommission the fork
